client-app/components/atoms/image-gallery/image-gallery.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <div
+    ref="mainImageDiv"
+    class="square relative flex flex-col justify-center items-center border border-gray-100 rounded-sm"
+  >
+    <img
+      :src="activeSrc"
+      alt="product.name"
+      class="absolute top-0 w-full h-full object-cover object-center rounded-sm"
+    />
+  </div>
+  <div v-if="isMobile && images && images.length > 1" class="mt-4 flex flex-row justify-center space-x-2.5">
+    <div
+      v-for="image in images"
+      :key="image?.url || ''"
+      class="border border-yellow-500 w-4 h-4 rounded-full cursor-pointer"
+      :class="{
+        'bg-yellow-500': image?.url == activeSrc,
+      }"
+      @click="setActiveImage(image?.url || '')"
+    ></div>
+  </div>
+  <div v-if="!isMobile && images && images.length > 1" class="mt-6 grid grid-cols-3 gap-5">
+    <div v-for="image in images" :key="image?.url || ''" @click="setActiveImage(image?.url || '')">
+      <div
+        class="square relative flex flex-col justify-center items-center cursor-pointer border border-gray-100 rounded-sm hover:ring hover:ring-yellow-200"
+        :class="{
+          'ring ring-yellow-500': image?.url == activeSrc,
+        }"
+      >
+        <img
+          :src="image?.url || '/static/images/no-image.png'"
+          alt="product.name"
+          class="absolute top-0 w-full h-full object-cover object-center rounded-sm"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ImageType, Maybe } from "@/core/api/graphql/types";
+import { onMounted, ref } from "vue";
+import { SwipeDirection, useSwipe } from "@vueuse/core";
+import _ from "lodash";
+
+const mainImageDiv = ref(null);
+
+useSwipe(mainImageDiv, {
+  onSwipeEnd: (e, direction) => {
+    if (direction !== SwipeDirection.LEFT && direction !== SwipeDirection.RIGHT && props.images.length < 2) {
+      return;
+    }
+    const activeImageIndex = _.findIndex(props.images, (x) => x?.url === activeSrc.value);
+
+    if (direction === SwipeDirection.LEFT && activeImageIndex < props.images.length - 1) {
+      const newActiveImageIndex = activeImageIndex + 1;
+      activeSrc.value = props.images[newActiveImageIndex]?.url ?? "";
+    }
+
+    if (direction === SwipeDirection.RIGHT && activeImageIndex > 0) {
+      const newActiveImageIndex = activeImageIndex - 1;
+      activeSrc.value = props.images[newActiveImageIndex]?.url ?? "";
+    }
+  },
+});
+
+const props = defineProps<{ src: string; images: Array<Maybe<ImageType>>; isMobile: boolean }>();
```

Comment:
Why props are defined in Generic, not in arguments? It's not readable at all here. Please, define props using full form with required/default props, etc.